frontend/src/components/ModelBox/ModelConfiguration.tsx

```diff
@@ -79,25 +86,18 @@ function ModelConfiguration({
 		});
 	}
 
-	// get model configs on mount
 	useEffect(() => {
-		chatService
-			.getGptModel()
-			.then((model) => {
-				// apply the currently set values
-				const newCustomChatModelConfigs = customChatModelConfigs.map(
-					(config) => {
-						const newConfig = { ...config };
-						newConfig.value = model.configuration[config.id];
-						return newConfig;
-					}
-				);
-				setCustomChatModel(newCustomChatModelConfigs);
-			})
-			.catch((err) => {
-				console.log(err);
-			});
-	}, []);
+		if (!chatModel) {
+			// chatModel is undefined if this is the first time that the user has switched to the sandbox level, and the change level request has not yet resolved successfully
+			return;
+		}
+		const newCustomChatModelConfigs = customChatModelConfigs.map((config) => {
+			const newConfig = { ...config };
+			newConfig.value = chatModel.configuration[config.id];
+			return newConfig;
+		});
```

Comment:
```
const newCustomChatModelConfigs = customChatModelConfigs.map((config) => ({
  ...config,
  value: chatModel.configuration[config.id],
});
```

For what it's worth, this is potentially wasteful because we are generating a brand new config object for each entry in customChatModelConfigs, even if it hasn't changed, which will trigger unnecessary renders. I seem to recall there's an issue to send and receive only the model config that the user has changed; if so, don't worry about it here.

Reply:
can't find the issue so I'll implement here

Reply:
There are no extra renders because `chatModel`, which we pass into this component, only changes on refresh of level change. Updating a config slider will not cause `chatModel` to update. I could change that (to keep them in sync), but I don't see it adding any functional value at this point. I want to avoid it because we'd have to drill another thing all the way down the components

Reply:
Yep ok, React is likely just optimising it based on keys.